Makefile

```diff
@@ -216,29 +227,34 @@ sync: git.pullmaster submodule-sync init
 
 .PHONY: submodule pull submodule-sync git.pullmaster
 
-# I tried to make this dependent on what I thought was the appropriate
-# lock file, but it caused the rule for that file to get run (which
-# seems to be about obtaining a new version of the 3rd party libraries).
-$(ISTIO_OUT)/istio_is_init: bin/init.sh istio.deps | ${ISTIO_OUT}
-	ISTIO_OUT=${ISTIO_OUT} bin/init.sh
-	touch $(ISTIO_OUT)/istio_is_init
-
-# init.sh downloads envoy
-${ISTIO_OUT}/envoy: init
-${ISTIO_ENVOY_DEBUG_PATH}: init
-${ISTIO_ENVOY_RELEASE_PATH}: init
-
-# Pull depdendencies, based on the checked in Gopkg.lock file.
-# Developers must manually call make depend.update if adding new deps or
-# make pull to pull recent changes in the submodule.
-depend: init | $(ISTIO_OUT)
+# The bare "envoy" files depend on istio.deps because their contents can depend on the
+# commit hash stores in that file.  The ISTIO_ENVOY_[RELEASE|DEBUG]_PATH contain the commit
+# sha that was read from istio.deps--they were computed from istio.deps and don't
+# need to depend on when the file was last updated (i.e., if the commit hash in istio.deps
+# changes then the path will update so we only care if the files exist or not).
+$(ISTIO_BIN)/envoy $(ISTIO_OUT)/envoy: istio.deps
+$(ISTIO_BIN)/envoy $(ISTIO_OUT)/envoy $(ISTIO_ENVOY_DEBUG_PATH): | ${FORTIO} ${ISTIO_OUT} ${ISTIO_BIN}
+	$(FORTIO) curl -L -stdclient -timeout 15m $(ISTIO_ENVOY_DEBUG_URL) > ${ISTIO_OUT}/envoy_debug.tar.gz
```

Comment:
I don't think I am comfortable with this piece. While we can consume our own dogfood, it does not have to become a prerequisite to bootstrap our code or build it.  The amount of options that curl/wget have is formidable. People building istio in inhouse infrastructures might be using custom options in curl or wget and they are more familiar with these than fortio.

Its fine to have fortio as a backup or for folks who are fans of fortio to use it instead of curl, but certainly not as the default way to download and build Istio.

Reply:
Let's be serious: we were ok to require java and Bazel as dependency, but we now need a backup for folks who can't install 'curl' ???

Reply:
I'm open to just expecting a working copy of curl.  I don't know how broadly people might have bought in to the fallbacks for no-curl or curl-without-https.  It just seemed odd to me to do this detection on third-party tools when istio itself already requires a programming language that already has built-in support to do an https download in a few lines of code that's smaller than the curl/wget detection logic.  I suppose I could do something like try curl and fallback to fortio (basically "curl || fortio") but I'm not sure if this would appease anyone.